frank-flow/src/frontend/src/app/flow/flow.component.ts

```diff
@@ -9,6 +16,65 @@ import { NodeService } from './node/node.service';
 export class FlowComponent {
   nodes = [];
   connections = [];
+  private panZoomConfigOptions: PanZoomConfigOptions = {
+    zoomLevels: 10,
+    zoomStepDuration: 0.2,
+    freeMouseWheelFactor: 0.01,
+    zoomToFitZoomLevelFactor: 0.5,
+    dragMouseButton: 'left',
+    zoomOnDoubleClick: false,
+  };
+  panzoomConfig: PanZoomConfig = new PanZoomConfig(this.panZoomConfigOptions);
 
-  constructor(private nodeService: NodeService) {}
+  @ViewChild('nodeContainer', { read: ElementRef })
+  nodeContainerRef!: ElementRef;
+  private offset = 500;
+
+  constructor(private renderer: Renderer2, private library: FaIconLibrary) {
+    this.library.addIcons(faArrowDown, faArrowUp, faArrowRight, faArrowLeft);
+  }
+
+  decreaseRight(): void {
+    const el = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0];
+    const elWidth = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0].offsetWidth;
+
+    this.renderer.setStyle(el, 'width', elWidth - this.offset + 'px');
+  }
+
+  decreaseBottom(): void {
+    const el = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0];
+    const elHeight = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0].offsetHeight;
+
+    this.renderer.setStyle(el, 'height', elHeight - this.offset + 'px');
+  }
+
+  expandRight(): void {
+    const el = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0];
+    const elWidth = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0].offsetWidth;
+
+    this.renderer.setStyle(el, 'width', elWidth + this.offset + 'px');
+  }
+
+  expandBottom(): void {
+    const el = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0];
+    const elHeight = this.nodeContainerRef.nativeElement.getElementsByClassName(
+      'canvas'
+    )[0].offsetHeight;
```

Comment:
A lot of duplicate code. Please get the canvas element first and reuse the same variable.
```
const el = this.nodeContainerRef.nativeElement.getElementsByClassName(
      'canvas'
    )[0];
    const elWidth = el.offsetWidth;
```

Reply:
Refactored.